### DISCUSSION TITLE
Supplying a vector parameter in Aptos Explorer

### DISCUSSION THREAD

**Discussion by Alexzander-Stone** [aptos-labs/aptos-developer-discussions#74] [Questions] [answered]
Asked by `mononok3` in Discord (#dev-discussion):

> How to enter vectors in explorer input?

**Keywords**
Aptos Explorer, input, arguments, vector

**Comment by Alexzander-Stone** [comment, marked as answer]
The values that you're supplying need to be wrapped in square brackets.

Here's an example of an input you could give for a vector<u8> on Aptos Explorer.

**Argument header**
`arg0: vector<u8>` 
**User input** 
`[97, 98, 99]`

**Reply by Alexzander-Stone** [reply]
To `monok3`, I can help more directly if you either supply the type of the vector you're interacting with or a link to the smart contract's entry function on the Aptos Explorer.

**Reply by huisq** [reply]
can we enter vector<vector<u6>>?

**Reply by Alexzander-Stone** [reply]
> can we enter vector<vector>?

Yes, you can!

Here's an example of an input for a `vector<vector<u8>>` on Aptos Explorer.

**Argument header**
`arg0: vector<vector<u8>>`
**User input**
`[[104, 101, 108, 108, 111, 95, 119, 111, 114, 108, 100], [104, 101]]`

----

Feel free to check out both examples on this [smart contract](https://explorer.aptoslabs.com/account/0xdf052910e29aaa24063e55a997aa6368e74587201b4c73ceab47bf20d1e9cf95/modules/code/aptos_explorer_vector?network=testnet) I deployed to the Testnet and test the above inputs.

**Reply by huisq** [reply]
<img width="684" alt="image" src="https://github.com/aptos-labs/aptos-developer-discussions/assets/50208058/464e20be-774a-43c4-8aab-99fcd796b89e">
getting this error

**Reply by Alexzander-Stone** [reply]
> getting this error

Looks like there's an issue on the Explorer side for elements represented as bigint (like u64). ~~I'll create an issue on the Explorer repo so attention can be put towards it~~ [A pull request](https://github.com/aptos-labs/explorer/pull/681) has been created that's an initial step to fixing the bug. 


### Temporary solution

For now, I'd recommend going through the Aptos CLI for sending these TXs. Here's [aptos.dev documentation](https://aptos.dev/move/move-on-aptos/cli#running-a-move-function) on running move functions from the CLI.

E.g. To perform the `takes_vector_of_vector_u64` on the testnet example [smart contract](https://explorer.aptoslabs.com/account/0xdf052910e29aaa24063e55a997aa6368e74587201b4c73ceab47bf20d1e9cf95/modules/code/aptos_explorer_vector?network=testnet), we can send a command like below:
```
aptos move run --function-id "devnet_dev_discussion::aptos_explorer_vector::takes_vector_of_vector_u64" \
    --args "u64:[[104, 101, 108, 108, 111, 95, 119, 111, 114, 108, 100], [104, 101]]" \
    --profile devnet_dev_discussion
```

**Reply by bokobza** [reply]
Hi @Alexzander-Stone, is there any way to do this from the Explorer itself?

**Reply by Alexzander-Stone** [reply]
Hi @huisq and @bokobza, the [PR](https://github.com/aptos-labs/explorer/pull/681/) I mentioned is live. 😌

Refresh you cache, and feel free to check it out against this [testnet example](https://explorer.aptoslabs.com/account/0xdf052910e29aaa24063e55a997aa6368e74587201b4c73ceab47bf20d1e9cf95/modules/run/aptos_explorer_vector/takes_vector_of_vector_u64?network=testnet).

**Reply by bokobza** [reply]
Hi @Alexzander-Stone , that's a quick turnover thanks.
What is the input supposed to look like for vector of vectors? I have an hex string, which the simple vector used to be able to transform to u8 but this doesn't seem to work anymore.
Still, I did a few tests and the results aren't as expected.
I'm expecting to see something similar to what I see here in the payload tab https://explorer.aptoslabs.com/txn/981112467/payload?network=testnet, so an array of numbers, but no matter whatever I try, I get something different.

1.
input: `["a11ceb0b06", "111ceb0b06"]`
transaction:
https://explorer.aptoslabs.com/txn/0x999b9283654d765eef7077eb18651d54ae01a8e5e4959ba6951ea8e9fcfe8cae/payload?network=testnet
result:
```
arguments:[
0:[
0:"0x61313163656230623036"
1:"0x31313163656230623036"]]
```
2.
input: `["0xa11ceb0b06", "0x111ceb0b06"]`
transaction: https://explorer.aptoslabs.com/txn/0x4f4ea3de80b7d3790a01329e7cb5ddde959423db53a6b1d713bc56b1cde75934/payload?network=testnet
result:
```
arguments:[
0:[
0:"0x307861313163656230623036"
1:"0x307831313163656230623036"]]
```

3.
input: `[["0xa11ceb0b06"], ["0x111ceb0b06"]]`
transaction: https://explorer.aptoslabs.com/txn/0xe78f35034e08b7675002f4da0a7e3e2100fd8c6b9804c7466a22a70cff3d0884/payload?network=testnet
result:
```
arguments:[
0:[
0:"0x00"
1:"0x00"]]
```

anything else is giving me an error in the console.

**Reply by Alexzander-Stone** [reply]
Since `takes_vector_of_vector` is expecting an input of `u8`, you're not able to supply hex strings on the Aptos Explorer. Instead, you'll need to convert your data into the u8 array representation.

It's possible we could add an automatic conversion, here... let me think on it for a sec. In the meanwhile, feel free to create an enhancement on the Aptos Explorer repo to expedite the feature @bokobza. Thanks for the heads up! 🫡

**Reply by bokobza** [reply]
Hi @Alexzander-Stone , I've been translating the hex to u8 in the meantime 🙏 
I'll see if I can make the explorer work. Thanks a lot!
